hack/changelog-check.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# Copyright 2020 the Velero contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# docker-push is invoked by the CI/CD system to deploy docker images to Docker Hub.
+# It will build images for all commits to master and all git tags.
+# The highest, non-prerelease semantic version will also be given the `latest` tag.
+
+set +x
+
+if [[ -z "$CI" ]]; then
+   echo "This script is intended to be run only on Github Actions." >&2
+   exit 1
+fi
+
+CHANGELOG_PATH='changelogs/unreleased'
+
+# https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request
+# GITHUB_REF is something like "refs/pull/:prNumber/merge"
+pr_number=$(echo $GITHUB_REF | cut -d / -f 3)
+
+change_log_file="${CHANGELOG_PATH}/${pr_number}-*"
+
+if ls ${change_log_file} 1> /dev/null 2>&1; then
+    echo "changelog for PR ${pr_number} exists"
+    exit 0
+else
+    echo "Please consider adding a changelog for PR ${pr_number}"
```

Comment:
Will this prevent merging? Or can we make it non-blocking?

Reply:
It is non-blocking.
Blocking/Non-Blocking is a setting in github itself, outside of this job.

Reply:
Ok, cool. I agree with non-blocking and using it as an easier check.

Reply:
It'd be a nice enhancement to make it required, but have it not fail if there's a `no-changelog-required` label on the PR (so we can explicitly opt out of requiring one for certain PRs). But, i'm not exactly sure how easy that is, and we can keep that in mind as a follow-up enhancement.

Reply:
Looks like labels are presented to Actions, since there's [this Action, which can check for required labels](https://github.com/marketplace/actions/verify-pull-request-labels). But I think it's also something we can do as an enhancement later.

Reply:
`if: contains(github.event.pull_request.labels.*.name, 'some-defined-label')` should be able to skip

Reply:
Yep, I was looking at something similar.
I'll do that as an enhancement.